components/application-mgt/org.wso2.carbon.identity.application.mgt/src/main/java/org/wso2/carbon/identity/application/mgt/ApplicationMgtUtil.java

```diff
@@ -915,20 +915,52 @@ private static User getUserFromTenant(String username, String userId, int tenant
      * Get user's tenant domain.
      *
      * @param tenantDomain The tenant domain which user is trying to access.
+     *                     This is the same tenant that application resides.
      * @param username     The username of the user.
      * @return The tenant domain where the user resides.
      * @throws IdentityApplicationManagementException Error when user cannot be resolved.
      */
     public static String getUserTenantDomain(String tenantDomain, String username)
             throws IdentityApplicationManagementException {
 
-        if (DOMAIN_QUALIFIED_REGISTRY_SYSTEM_USERNAME.equals(username) ||
+        try {
+            if (useApplicationTenantDomainAsUserTenantDomain(tenantDomain, username)) {
+                return tenantDomain;
+            }
+            Optional<User> user = getUser(tenantDomain, username);
+            if (!user.isPresent()) {
+                /*
+                This situation occur when the application creator is deleted. At that point,
+                 set the tenant domain of the application as the user's tenant domain.
+                 */
+                return tenantDomain;
+            }
+            return user.get().getTenantDomain();
```

Comment:
```suggestion
            /*
             Else situation occur when the application creator is deleted. At that point,
             set the tenant domain of the application as the user's tenant domain.
             */
            return getUser(tenantDomain, username).map(User::getTenantDomain).orElse(tenantDomain);
   
```

Reply:
fixed in [8db456c](https://github.com/wso2/carbon-identity-framework/pull/4302/commits/8db456c974f555f629f9695bce1d362bdedc277f)